lib/tapioca/dsl/compilers/active_model_validations_confirmation.rb

```diff
@@ -0,0 +1,97 @@
+# typed: strict
+# frozen_string_literal: true
+
+begin
+  require "active_model"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Dsl
+    module Compilers
+      # `Tapioca::Dsl::Compilers::ActiveModelValidationsConfirmation` decorates RBI files for all
+      # classes that use [`ActiveModel::Validates::Confirmation`](https://api.rubyonrails.org/classes/ActiveModel/Validations/HelperMethods.html#method-i-validates_confirmation_of).
+      #
+      # For example, with the following class:
+      #
+      # ~~~rb
+      # class User
+      #   include ActiveModel::Validations
+      #
+      #   validates_confirmation_of :password
+      #
+      #   validates :email, confirmation: true
+      # end
+      # ~~~
+      #
+      # this compiler will produce an RBI file with the following content:
+      # ~~~rbi
+      # # typed: true
+      #
+      # class User
+      #
+      #   sig { returns(T.untyped) }
+      #   def email_confirmation; end
+      #
+      #   sig { params(email_confirmation=: T.untyped).returns(T.untyped) }
+      #   def email_confirmation=(email_confirmation); end
+      #
+      #   sig { returns(T.untyped) }
+      #   def password_confirmation; end
+      #
+      #   sig { params(password_confirmation=: T.untyped).returns(T.untyped) }
+      #   def password_confirmation=(password_confirmation); end
+      # end
+      # ~~~
+      class ActiveModelValidationsConfirmation < Compiler
+        extend T::Sig
+
+        ConstantType = type_member do
+          {
+            fixed: T.all(
+              T::Class[ActiveModel::Validations],
+              ActiveModel::Validations::HelperMethods,
+              ActiveModel::Validations::ClassMethods,
+            ),
+          }
+        end
+
+        class << self
+          sig { override.returns(T::Enumerable[Module]) }
+          def gather_constants
+            # Collect all the classes that include ActiveModel::Validations
+            all_classes.select do |c|
+              c < ActiveModel::Validations &&
+                c.name.to_s != "ActiveRecord::Base"
+            end
+          end
+        end
+
+        sig { override.void }
+        def decorate
+          confirmation_validators = constant.validators.select do |v|
+            v.is_a?(ActiveModel::Validations::ConfirmationValidator)
+          end
+
+          return if confirmation_validators.empty?
+
+          # Create a RBI definition for each class that includes Active::Model::Validations
+          root.create_path(constant) do |klass|
+            # Create RBI definitions for all the attributes that use confirmation validation
+            confirmation_validators.each do |validator|
+              validator.attributes.each do |attr_name|
+                klass.create_method("#{attr_name}_confirmation", return_type: "T.untyped")
```

Comment:
Would it be possible to figure out the right types based on the columns?

Reply:
If the it's a subclass of ActiveRecord::Base, we can probably use one of ActiveRecord's API to figure out the column type and translate it to a proper return type. The same could be said for classes that implement ActiveModel::Attributes. 

However, since this validations can be added to POROs, we'd have to add logic to differentiate the two. Are we ok with doing that here?

Reply:
Good point. Do these validations require you to use Active Model attributes or something that would allow us to fetch the type?

If not, I think it's fine to use `T.untyped`. Just trying to think if there's a way to get better typing support.